### ISSUE TITLE
Cronjob not importing data

### ISSUE THREAD

**Issue by Daniel-McLaughlan** [campai-bv/autotask-dashboards-os#3]
Hi!

After upgrading to 1.3.2 earlier this week, our dashboards seemed to be running well and importing correctly and displaying accurate data. 

However yesterday afternoon the dashboards stopped displaying the widgets.. 

I've checked cronjob.log and found an error:

".. Could not delete view cache!"
"Failed: We've encountered some errors while running the import script."

debug.log and error.log are returning a variety of errors as well. 

I've recreated bootstrap.php from the original which I had backed up, after reading the attached error (error.log) which I believe implies it can't open bootstrap.php, however the issue persists. 

Any idea why this might be occurring? 
Is it worth debugging this or now that I know what I'm doing is it probably easier to just rebuild from scratch?

![campai errorlog](https://f.cloud.github.com/assets/5236131/968115/520ded84-0594-11e3-9f27-e410e4d1de0d.PNG)

**Comment by coencoppens** [comment]
Hi Daniel,

It's definitely worth figuring out what's going wrong here, since there might be others having the same issue.
- Did you update your bootstrap with the Cache configuration (see the default)?
- What happens when you manually remove all the cache?

``` text
rm -f /var/www/app/tmp/cache/views/* /var/www/app/tmp/cache/persistent/* /var/www/app/tmp/cache/models/*
```
- What are your error.log and debug.log saying?
- Is this a duplicate of issue #4?

**Comment by jarrodmast** [comment]
that almost looks like some sort of error in the bootstrap file?
would be best to post copy of bootstrap and the debug/error logs
Issue #4 doesn't have any other side effects other than breaking the cronjob log

**Comment by Daniel-McLaughlan** [comment]
Hi Guys,

Sorry for the long response on this one, I have spent the last week out of the office and have not had a chance to look at this. 

I had a look this morning and have re-configured my bootstrap.php file as per issue #4 , which appears to have removed most of my existing errors.

The issue I have now is that there is still no data being displayed in the interface, despite absolutely everything suggesting that the data is being imported. As per the attached Image, you can see that it is 12:05 PM and the last import ran at 12:00. 
![campai 26082013](https://f.cloud.github.com/assets/5236131/1023533/3b7f05b8-0df4-11e3-8885-b67b0b8a06cb.png)

The logs appear to be pretty clean and the cronjob.log is returning that it has imported correctly, the only thing i'm not sure about is the line in error.log: 

2013-08-26 11:58:12 Error: [NotFoundException] Debug setting does not allow access to this url.
Stack Trace:
#0 {main}

I have restarted the apache2 service and manually run the cronjob to try and pull in current ticket data. I have also edited my bootstrap.php file to only import tickets from the past 100 days.

Any suggestions as to why it appears as though it's importing data but nothing's showing up in the interface?

**Comment by MattParr** [comment]
I've seen that blank dashboard when the cache directory structure is missing.
You should have this structure writable by the web server:

app/tmp/cache/models
app/tmp/cache/persistent
app/tmp/cache/views
app/tmp/logs
app/tmp/sessions
app/tmp/tests

**Comment by Daniel-McLaughlan** [comment]
I just rebuilt the server from scratch and have found myself with the same problem again. 

I installed the server, updated bootstrap.php as per https://github.com/coencoppens/autotask-dashboards/blob/master/README (and included cache configuration as mentioned in issued #4  )

Then upgraded from 1.1.1 to 1.2.0, imported data. Deleted caches. ( rm -f /var/www/app/tmp/cache/persistent/\* /var/www/app/tmp/cache/models/\* )

Then upgraded to 1.3.0 from 1.2.0, then upgraded to 1.3.2 from 1.3.0 using the 1.3.0-to-1.3.2..sql update. 

I have imported all my ticket status'/queues etc. but when I create a dashboard (I didn't yet go to the effort of naming queues etc.), even if I select every measurable item, the dashboard comes up blank, the same as the last screenshot. 

What am I doing wrong here?

I have read through both te CAMAPI Documentation (http://www.autotask.campai.nl/documentation/how-to-upgrade-your-vm/) as well as the README's located in GITHUB and confirmed all settings reflect those mentioned. 

I took a VM snapshot at 1.2.0 and have just rolled back to 1.2.0, confirming that at this point in time the dashboard is functional. 

From here, I downloaded, extracted and ran 1.3.2 upgrade _Using only 1.2.0 to 1.3.2 upgrade.)

When I tried to run an import job, the following error was returned in the console: 
. 
![campai console](https://f.cloud.github.com/assets/5236131/1046551/04f40cec-1041-11e3-8b37-9ee1eed5b154.png)

**Comment by coencoppens** [comment]
@Daniel-McLaughlan : Thanks for the clear debugging.

First of all, please update to latest [release v1.3.3](https://github.com/coencoppens/autotask-dashboards/releases/tag/v1.3.3) since that will solve potential problems.

Now about your issue: the error log tells you there are missing a table in your database:

``` text
Table timeentries for model Timeentry was not found in
```

So if you execute this query you should be fine:

``` sql
CREATE TABLE `timeentries` (
  `id` int(10) NOT NULL AUTO_INCREMENT,
  `created` datetime NOT NULL,
  `resource_id` int(10) NOT NULL,
  `ticket_id` int(10) NOT NULL,
  `hours_to_bill` double(10,2) NOT NULL,
  `hours_worked` double(10,2) NOT NULL,
  `non_billable` tinyint(1) NOT NULL,
  `offset_hours` double(10,2) NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```

When running it again you might have other missing table showing up. You can find all queries in

``` text
/app/Plugin/Autotask/Config/Schema/autotask-1.3.0.sql
```

Does this solve the issue?

**Comment by Daniel-McLaughlan** [comment]
Can you advise of the order of updating?

Say i'm starting at 1.2.0 (where my last working VM snapshot is) should I update straight to 1.3.2, then to 1.3.3?

Update: 30 minutes later

I have updated from 1.2.0 to 1.3.2, then to 1.3.3 and all appears to be functioning correctly except for one error in cronjob.log  -  <notice>2013-08-29 10:01:01 Notice: Undefined property: stdClass::$StartDateTime in [/var/www/app/Plugin/Autotask/Console/Command/Task/CalculateTotalsForTimeEntriesTask.php, line 85]
</notice>

Cheers.

**Comment by jarrodmast** [comment]
Thats great @Daniel-McLaughlan this is a bug @coencoppens fixed in 1.4.0 branch as part of issue #8 changes,
see commit https://github.com/coencoppens/autotask-dashboards/commit/3e8ac81454ee00c10293f9566da6c29b1640f6c4 

basically pop this code right before the `$this->Timeentry->create();` line

``` php
          if( isset( $oResult->StartDateTime ) ) {
            $sStartDateTime = $oResult->StartDateTime;
          } else {
            $sStartDateTime = $oResult->CreateDateTime;
          }
```

and replace `$oResult->StartDateTime` with `$sStartDateTime` on the 'created' line

**Comment by coencoppens** [comment]
As @jarrodmast said, this will be fixed in the next release :-)
